src/webgpu/shader/validation/wgsl/duplicate-let-name-within-func.fail.wgsl

```diff
@@ -0,0 +1,11 @@
+// v-0014 - This fails because constant `a` is redeclared.
+
+[[stage(fragment)]]
+fn main() {
+  let a = 1u;
+  if (true) {
+    let a = 2u;
```

Comment:
This is perfectly valid in WGSL; inter-scope shadowing is supported.

The key spec language is this (emphasis mine):
> A declaration must not introduce a name when that identifier is already in scope **with the same end scope** as another instance of that name.

The end scope of the second `a` declaration is not the same as the first `a` declaration, and so this is valid.

Reply:
Thanks, seems that your clarification raised another problem with `var` tests. We can return to `let` little bit later.